engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoPolyglotJava.java

```diff
@@ -158,26 +173,37 @@ static void close(EnsoContext ctx) {
   private Object findPolyglotJava() throws InteropException {
     TruffleSafepoint.setBlockedThreadInterruptible(null, Semaphore::acquire, lock);
     try {
-      if (polyglotJava != this) {
-        return polyglotJava;
-      }
-      polyglotJava = createPolyglotJava(ctx);
-      while (!pendingPath.isEmpty()) {
-        InteropLibrary.getUncached()
-            .invokeMember(polyglotJava, "addPath", pendingPath.remove(0).toString());
-      }
-      try {
-        InteropLibrary.getUncached()
-            .invokeMember(polyglotJava, "findLibraries", new LibraryResolver());
-      } catch (InteropException ex) {
-        logger.log(Level.WARNING, "Cannot register findLibraries", ex);
+      if (polyglotJava == this) {
+        polyglotJava = createPolyglotJava(ctx);
+        try {
+          InteropLibrary.getUncached()
+              .invokeMember(polyglotJava, "findLibraries", new LibraryResolver());
+        } catch (InteropException ex) {
+          logger.log(Level.WARNING, "Cannot register findLibraries", ex);
+        }
       }
+      adjustClassPath();
```

Comment:
- picking those entries added to `classPath` up when `findPolyglotJava` is requested 
- at such a moment we know we _can use Truffle lock_